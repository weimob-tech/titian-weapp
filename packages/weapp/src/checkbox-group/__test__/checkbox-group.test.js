const simulate = require('miniprogram-simulate');
const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', () => {
    const comp = getComponent('checkbox-group');
    comp.triggerLifeTime('attached');
    expect(comp).toMatchSnapshot();
  });

  it('should render options props success', async () => {
    const options = [
      { value: 'a', label: '选项 A' },
      { value: 'b', label: '选项 B' },
      {
        value: 'c',
        label: '选项 C'
      }
    ];
    const comp = getComponent('checkbox-group', {
      options,
      max: 1,
      name: 'test',
      value: ['a', 'c']
    });

    comp.triggerLifeTime('attached');
    await simulate.sleep(300);
    expect(comp).toMatchSnapshot();
  });
});
