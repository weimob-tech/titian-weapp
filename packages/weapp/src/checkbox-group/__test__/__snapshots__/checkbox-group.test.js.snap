// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render options props success 1`] = `
<main
  name="test"
>
  <wx-view
    class="ext-class titian-checkbox-group"
    style=""
  >
    <ti-checkbox
      class="internal_children"
    >
      <wx-view
        class="ext-class titian-checkbox-checked titian-checkbox"
        style=""
      >
        <wx-view
          class="titian-checkbox-icon-wrap-checked titian-checkbox-icon-wrap"
          style="--base-radius-size:999px; height: 32rpx"
          bind:tap="onChange"
        >
          <ti-icon
            extClass="icon-class titian-checkbox-icon-zoom-in titian-checkbox-icon-checked titian-checkbox-icon-animation titian-checkbox-icon"
          >
            <wx-view
              class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
              style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
              bind:tap="onClick"
            />
          </ti-icon>
        </wx-view>
        <wx-view
          class="titian-checkbox-label"
          bind:tap="onClick"
        >
          选项 A
        </wx-view>
      </wx-view>
    </ti-checkbox>
    <ti-checkbox
      class="internal_children"
    >
      <wx-view
        class="ext-class titian-checkbox"
        style=""
      >
        <wx-view
          class="titian-checkbox-icon-wrap"
          style="--base-radius-size:999px; height: 32rpx"
          bind:tap="onChange"
        >
          <ti-icon
            extClass="icon-class titian-checkbox-icon-zoom-out titian-checkbox-icon-animation titian-checkbox-icon"
          >
            <wx-view
              class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
              style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
              bind:tap="onClick"
            />
          </ti-icon>
        </wx-view>
        <wx-view
          class="titian-checkbox-label"
          bind:tap="onClick"
        >
          选项 B
        </wx-view>
      </wx-view>
    </ti-checkbox>
    <ti-checkbox
      class="internal_children"
    >
      <wx-view
        class="ext-class titian-checkbox-checked titian-checkbox"
        style=""
      >
        <wx-view
          class="titian-checkbox-icon-wrap-checked titian-checkbox-icon-wrap"
          style="--base-radius-size:999px; height: 32rpx"
          bind:tap="onChange"
        >
          <ti-icon
            extClass="icon-class titian-checkbox-icon-zoom-in titian-checkbox-icon-checked titian-checkbox-icon-animation titian-checkbox-icon"
          >
            <wx-view
              class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
              style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
              bind:tap="onClick"
            />
          </ti-icon>
        </wx-view>
        <wx-view
          class="titian-checkbox-label"
          bind:tap="onClick"
        >
          选项 C
        </wx-view>
      </wx-view>
    </ti-checkbox>
  </wx-view>
</main>
`;

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="ext-class titian-checkbox-group"
    style=""
  />
</main>
`;
